cpp/src/rolling/detail/range_utils.cuh

```diff
@@ -656,26 +674,199 @@ struct range_window_clamper {
     };
     if (grouping.has_value()) {
       if (orderby.has_nulls()) {
-        copy_n(distance_kernel{grouped_with_nulls{nulls_at_start,
-                                                  grouping->labels.data(),
-                                                  grouping->offsets.data(),
-                                                  grouping->nulls_per_group.data()},
-                               d_row_delta,
-                               d_begin,
-                               d_end});
+        if (Direction == direction::PRECEDING) {
```

Comment:
All the changes in this block are an attempt to test removing template parameters from the `range_window_clamper` so that only the `distance_kernel` is templated (and we can follow up with benchmarks of removing templates from the kernel). There are no material changes here.